### PR TITLE
HCAP-1494/1495 - adjust queries for new role and add search filter for program

### PR DESCRIPTION
**Pull request by coryatwater-FW**
### Pending
- update tests

### Summary
- encompasses HCAP-1494 and HCAP-1495, updating backend with new role and program field, as well as adding filter by program functionality to frontend

### Backend
- adds `program` to participant queries
- update existing queries with new `mhsu-employer` role
- add new `programFilter` depending on role
- add program filter to participant query param

### Frontend
- add filter by program functionality

### AC
- Regression: queries work as usual
- The participants should be returned as expected across PE / HA / MoH 
- HAs / MoHs can filter all HCA participants
- HAs / MoHs can filter all MHSU participants
- HAs / MoHs can have no filter and see all participants
- Private Employers can see only HCA participants
- MHSU Employers can see only MHSU participants
- If a MHSU Employer visits the link to a participant details page of an HCA Participant, they should see an error and no - information. Vice versa.

### CODE REVIEW

File: client/src/pages/private/ParticipantTable.js [hunk 1]
```diff
@@ -365,14 +370,25 @@ const ParticipantTable = () => {
...
+
+    if (isMoH || isHA || isSuperUser) {
+      // should be able to see all users in both MHAW/ HCA programs
+      setPrograms(allPrograms);
+    } else if (isEmployer) {
+      // private employers should only see HCA program
+      setPrograms(programsPrivateEmployer);
+    } else if (isMHSUEmployer) {
+      // MHAW employers should only see MHAW program
+      setPrograms(programsMHAWEmployer);
```
Comment (coryatwater-FW) [inline client/src/pages/private/ParticipantTable.js:389]: will change this to stay consistent with MHSU

File: client/src/constants/participantTableConstants.js [hunk 2]
```diff
@@ -22,23 +22,23 @@ export const participantEngageStatus = {
 export const tabs = {
   // Tabs, associated allowed roles, displayed statuses
   'Available Participants': {
-    roles: [Role.Employer, Role.HealthAuthority],
+    roles: [Role.Employer, Role.MHSUEmployer, Role.HealthAuthority],
```
Comment (sunghwan-park-fw) [inline client/src/constants/participantTableConstants.js:25]: This array could be defined as a constant.

File: client/src/pages/private/ParticipantTableFilters.js [hunk 3]
```diff
@@ -62,7 +62,31 @@ export const ParticipantTableFilters = ({ loading, locations }) => {
...
+            inputProps={{ displayEmpty: true }}
+            disabled={loading || programs.length === 1}
+            value={filter[FILTERABLE_FIELDS.PROGRAM]?.value || ''}
+            onChange={({ target }) => setFilter(FILTERABLE_FIELDS.PROGRAM, target.value)}
+            aria-label='program filter'
+          >
+            {programs.length === 1 ? (
+              <MenuItem value=''>{programs[0]}</MenuItem>
+            ) : (
+              ['Program', ...programs].map((option, index) => (
```
Comment (sunghwan-park-fw) [inline client/src/pages/private/ParticipantTableFilters.js:80]: When I first saw the 'Program' option, I wondered what it meant. How about 'All'?

**Review by sunghwan-park-fw — APPROVED** [review]
I left a few comments but it looks good.

File: client/src/pages/private/ParticipantTableFilters.js (hunk 3 above)
Comment (coryatwater-FW) [reply client/src/pages/private/ParticipantTableFilters.js:80]: it's basically acting as the label/ placeholder for what filter it is as there's no actual label describing it

**Review by sunghwan-park-fw — APPROVED** [review]

**Pull request merged**